sympy/assumptions/handlers/sets.py

```diff
@@ -770,3 +770,26 @@ def _(expr, assumptions):
     x = expr.args[0]
     if ask(Q.algebraic(x), assumptions):
         return ask(~Q.nonzero(x - 1), assumptions)
+
+
+# TranscendentalPredicate
+
+@TranscendentalPredicate.register_many(Exp1, Pi, GoldenRatio)
+def _(expr, assumptions):
+    return True
+
+@TranscendentalPredicate.register_many(AlgebraicNumber, Float, TribonacciConstant,
+    Infinity, ImaginaryUnit, ComplexInfinity)
+def _(expr, assumptions):
+    return False
+
+@TranscendentalPredicate.register(Expr)
+def _(expr, assumptions):
+    ret = expr.is_transcendental
```

Comment:
I see, so this was just missing for transcendental. Let's make sure that it's not missing for any other assumptions as well.

Reply:
Let's add an explicit test that this works. Right now pi and E will just use the above handlers, but we can add a test that `ask(Q.transcendental(Symbol('x', transcendental=True)))` works.

Reply:
Let me check explicitly each assumptions. I'll add the same changes if not handled properly. Should I create a separate function to test the above in `test_query.py`?

Reply:
I've checked for other predicats, I think they are handled properly as is. That's also reflected in their test cases.